src/sign/tests.rs

```diff
@@ -1177,6 +1177,65 @@ fn test_sign_roundtrip() {
         .is_err());
 }
 
+#[test]
```

Comment:
Could we add tests that check what happens when trying to create/verify a detached signature for a COSE object that has a non-`nil` payload?

Reply:
Added tests for all helper methods that verify that they panic when self.payload.is_some().